### PR TITLE
Fixed issue which was causing problem while testing for headings challenge

### LINKED ISSUE #17833: Challenge(s) doesn't verify proper closing tags
#### Describe your problem and - if possible - how to reproduce it
The challenge **"Applied Accessibility: Use Headings to Show Hierarchical Relationships of Content"** (and possibly other challenges as well) doesn't verify proper tag closing.

To reproduce: Replace all `<h5>` tags with `<h3>` tags, but leave the `</h5>` tags as they are. The challenge will pass.

#### Add a Link to the page with the problem
https://learn.freecodecamp.org/responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content

#### Tell us about your browser and operating system

* Browser Name: Firefox Quantum
* Browser Version: 61.0
* Operating System: Windows 7 (64 bit)

#### If possible, add a screenshot here

afbeelding

### PR DESCRIPTION
**Pull request by thecodingaviator**
@QuincyLarson / @raisedadead / @Bouncey can anyone of you check that my code works? I'm not able to test it right now because for some funny reason, it is working with my RegEx tester but not on a jQuery playground. Please check

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #17833

### CODE REVIEW

**Comment by thecodingaviator** [comment]
@Asjas

**Comment by thecodingaviator** [comment]
@Asjas

**Comment by thecodingaviator** [comment]
@scissorsneedfoodtoo Can you have a look at this PR?

**Comment by ezioda004** [comment]
@thecodingaviator Can you fix the conflict? I'll test it locally.

**Comment by thecodingaviator** [comment]
> @thecodingaviator Can you fix the conflict? I'll test it locally.

@ezioda004 Done

**Comment by ezioda004** [comment]
The challenge seems to be missing locally for me.

Can someone else check it out?
cc: @nikrb

**Review by nikrb — DISMISSED** [review]
I think the quotes needs sorting out as they seem to be causing the build (bootstrap) to fail.

**Comment by nikrb** [comment]
@ezioda004 
> expected [Function] to not throw an error

we don't have a function - so I added `() =>` and tests seem to pass now. Do you know why the test strings are wrapped in single quotes? With or without doesn't seem to make a difference, in this particular challenge atleast.

Looking at other challenges, some use `assert( getUserInput => {});` I wonder if that is best practice? Really don't *know* how it is supposed to work so will have a play later.

I'll test the app and push the changes if everything looks good.

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.english.md [hunk 1]
```diff
@@ -28,8 +28,11 @@ tests:
   - text: Your code should have six <code>h3</code> tags.
     testString: assert($('h3').length === 6, 'Your code should have six <code>h3</code> tags.');
   - text: Your code should not have any <code>h5</code> tags.
-    testString: assert($('h5').length === 0, 'Your code should not have any <code>h5</code> tags.');
-
+    testString: 'assert($("h5").length === 0, "Your code should not have any <code>h5</code> tags.");'
+  - text: Your code should not have any <code>h5</code> closing tags.
+    testString: 'assert({var temp = document.getElementById('html-seed').innerHTML; var count = (temp.match(/\/h5/g) || []).length; count===0}, "Your code should not have any <code>h5</code> closing tags.");'
+  - text: Your code should not have 6 <code>h3</code> closing tags.
+    testString: 'assert({var temp = document.getElementById('html-seed').innerHTML; var count = (temp.match(/\/h3/g) || []).length; count===6}, "Your code should not have 6 <code>h3</code> closing tags.");'
```
[the remark this answers is not in the document]
Comment (thecodingaviator) [reply curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.english.md]: Done

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.english.md [hunk 2]
```diff
@@ -28,8 +28,11 @@ tests:
   - text: Your code should have six <code>h3</code> tags.
     testString: assert($('h3').length === 6, 'Your code should have six <code>h3</code> tags.');
   - text: Your code should not have any <code>h5</code> tags.
-    testString: assert($('h5').length === 0, 'Your code should not have any <code>h5</code> tags.');
-
+    testString: 'assert($("h5").length === 0, "Your code should not have any <code>h5</code> tags.");'
+  - text: Your code should not have any <code>h5</code> closing tags.
+    testString: 'assert({var temp = document.getElementById('html-seed').innerHTML; var count = (temp.match(/\/h5/g) || []).length; count===0}, "Your code should not have any <code>h5</code> closing tags.");'
```
[the remark this answers is not in the document]
Comment (thecodingaviator) [reply curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-headings-to-show-hierarchical-relationships-of-content.english.md]: Done

**Comment by thecodingaviator** [comment]
> @ezioda004 had some trouble getting the tests to work in the app. Ended up with `assert((code.match(/\/h3/g) || []).length===6,'Your code should have 6 <code>h3</code> closing tags.');`
> Can you test locally to make sure this works for you?

You're asking me to test? If so, I cannot, haven't setup a local build

**Comment by nikrb** [comment]
@thecodingaviator no I was asking @ezioda004 if he could test it locally :)

**Review by ezioda004 — COMMENTED** [review]
@nikrb Successfully tested locally, it seems to work.
>we don't have a function - so I added () => ...

Am I missing something? I dont see the callback function in the tests.
Also, from my experience I've used a function for the tests which needs statements that cannot be expressed in a single line. Which seems to be what most of the other tests have been doing.

Coming to the issue I have with the instructions - 
>Camper Cat wants a page on his site dedicated to becoming a ninja. Help him fix the headings so his markup gives semantic meaning to the content, and shows the proper parent-child relationships of his sections. Change all the h5 tags to the proper heading level to indicate they are subsections of the h2 ones.

There is no mention of `<h3>` tags in the instructions. It'd be better if we explicitly mention it. Otherwise the only way to know right now is through the tests.

**Comment by thecodingaviator** [comment]
I feel what @ezioda004 is saying is right. So I have explicitly mentioned that the user should use h3 tags

**Comment by thecodingaviator** [comment]
Closed/reopened to clear Travis CI error

**Comment by nikrb** [comment]
@ezioda004 successfully tested locally after latest changes. I think we need someone else to test it locally and travis needs a prodding.

**Comment by Nirajn2311** [comment]
@ezioda004 Ill test it out now.

**Review by Nirajn2311 — APPROVED** [review]
@thecodingaviator Everything LGTM, just could you make `6` to `six` or vice versa as I don't think it will be proper if one is in words and the other in numbers.
@ezioda004 @nikrb All tests have passed.

**Comment by thecodingaviator** [comment]
@Nirajn2311 done

**Review by ezioda004 — APPROVED** [review]

**Comment by ezioda004** [comment]
Thank you for your contribution.

**Pull request merged**